### ISSUE TITLE
rustls 0.21.4 breaks compilation of async-nats 0.29.0

### ISSUE THREAD

**Issue by seb-sky** [nats-io/nats.rs#1017]
Using rustls v0.21.4 (released 1h ago):

```
   Compiling async-nats v0.29.0
error[E0282]: type annotations needed
  --> /[...]/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.29.0/src/tls.rs:69:20
   |
69 |         root_store.add_parsable_certificates(
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `C` declared on the method `add_parsable_certificates`

For more information about this error, try `rustc --explain E0282`.
error: could not compile `async-nats` (lib) due to previous error
```

Note that rustls v0.21.3 works fine.

**Comment by ctz** [comment]
Sorry about this. https://doc.rust-lang.org/cargo/reference/semver.html#minor-generalizing-a-function-to-use-generics-supporting-original-type suggested the change we made upstream was semver minor (but then goes on to say "Introducing generics in this way can potentially create type inference failures" which is not super helpful for a document that is meant to give advice on what changes have which semver properties).

I think our expectation was that callers would be with `Vec<u8>` or `&[u8]` which both satisfy `IntoIterator<Item = AsRef<[u8]>>`, but naturally the previous signature allowed more than those two options to exist and so this change was more breaking than we expected.

**Comment by n1ghtmare** [comment]
The rustls team issued a hotfix and it should now be resolved with rustls `v0.21.5`. https://github.com/rustls/rustls/issues/1352

**Comment by Jarema** [comment]
rustls reverted the breaking change

https://github.com/rustls/rustls/releases/tag/v%2F0.21.5

Library builds fine again.

Closing.
